src/main/java/de/metas/ui/web/handlingunits/process/WEBUI_M_HU_MoveToAnotherWarehouse.java

```diff
@@ -46,76 +28,22 @@
  * #L%
  */
 
-/**
- * #2144
- * HU editor: Move selected HUs to another warehouse
- *
- * @author metas-dev <dev@metasfresh.com>
- *
- */
-public class WEBUI_M_HU_MoveToAnotherWarehouse extends HUEditorProcessTemplate implements IProcessPrecondition
+public class WEBUI_M_HU_MoveToAnotherWarehouse extends WEBUI_M_HU_MoveToAnotherWarehouse_Helper
 {
-	private final transient IHUMovementBL huMovementBL = Services.get(IHUMovementBL.class);
-	private final transient IHandlingUnitsDAO handlingUnitsDAO = Services.get(IHandlingUnitsDAO.class);
-	private final transient IHUStatusBL huStatusBL = Services.get(IHUStatusBL.class);
-
-	@Param(parameterName = I_M_Warehouse.COLUMNNAME_M_Warehouse_ID, mandatory = true)
-	private I_M_Warehouse warehouse;
-
-	private HUMovementResult movementResult = null;
+	private final transient HULotNumberLockService lotNumberLockService = Adempiere.getBean(HULotNumberLockService.class);
 
 	@Override
 	protected ProcessPreconditionsResolution checkPreconditionsApplicable()
 	{
-		if (!isHUEditorView())
-		{
-			return ProcessPreconditionsResolution.rejectWithInternalReason("not the HU view");
-		}
-
-		if (!streamSelectedHUIds(Select.ONLY_TOPLEVEL).findAny().isPresent())
-		{
-			return ProcessPreconditionsResolution.reject(msgBL.getTranslatableMsgText(WEBUI_HU_Constants.MSG_WEBUI_ONLY_TOP_LEVEL_HU));
-		}
-
-		final boolean activeTopLevelHUsSelected = streamSelectedHUs(Select.ONLY_TOPLEVEL)
-				.filter(huRecord -> huStatusBL.isPhysicalHU(huRecord))
+		final boolean lockedHUs = streamSelectedHUs(Select.ONLY_TOPLEVEL)
+				.filter(huRecord -> lotNumberLockService.isLockedHU(huRecord))
 				.findAny()
 				.isPresent();
-		if (!activeTopLevelHUsSelected)
+		if (lockedHUs)
 		{
-			return ProcessPreconditionsResolution.rejectWithInternalReason("no physical (etc active) hus selected");
+			return ProcessPreconditionsResolution.rejectWithInternalReason("Some HUs are locked");
```

Comment:
please add a i18nized text, so the user doesn't have to search the process, but knows why they can't use it